### PR TITLE
Group norm fp16 support

### PR DESCRIPTION
**Pull request by wwbitejotunn**
### PR types
Function optimization

### PR changes
OPs

### Describe
Add fp16 support for group norm kernel and trt group norm plugin. The group norm fp16 mode works well in stable diffusion and disco diffusion. The accuracy of group norm fp16 is rtol=1e-2, atol=1e-3 in unit test.

### CODE REVIEW

File: paddle/fluid/inference/tensorrt/plugin/group_norm_op_plugin.cu [hunk 1]
```diff
@@ -70,48 +116,51 @@ int GroupNormPlugin::enqueue(int batch_size,
...
-             groups_,
-             eps_,
-             output,
-             mean_d,
-             variance_d,
-             DataLayout::kNCHW);
+  float *mean_d = static_cast<float *>(workspace);
+  float *variance_d = mean_d + input_shape[0] * groups_;
+  float *temp_variance_d = variance_d + input_shape[0] * groups_;
+  // phi::DenseTensor mean_t;
```
Comment (MARD1NO) [inline paddle/fluid/inference/tensorrt/plugin/group_norm_op_plugin.cu]: 这个是不是直接删了

File: paddle/phi/kernels/gpu/group_norm_kernel.cu [hunk 2]
```diff
@@ -283,43 +286,60 @@ void GroupNormDirectCUDAFunctor<T>::operator()(gpuStream_t stream,
...
 
-PD_REGISTER_KERNEL(
-    group_norm, GPU, ALL_LAYOUT, phi::GroupNormKernel, float, double) {}
+PD_REGISTER_KERNEL(group_norm,
+                   GPU,
+                   ALL_LAYOUT,
+                   phi::GroupNormKernel,
+                   float,
+                   double,
+                   phi::dtype::float16) {}
```
Comment (yuanlehome) [inline paddle/phi/kernels/gpu/group_norm_kernel.cu:345]: 建议在pr描述里说明一下性能和精度数据

Comment (wwbitejotunn) [reply paddle/phi/kernels/gpu/group_norm_kernel.cu:345]: Done, thx

File: paddle/fluid/inference/tensorrt/plugin/group_norm_op_plugin.cu (hunk 1 above)
Comment (wwbitejotunn) [reply paddle/fluid/inference/tensorrt/plugin/group_norm_op_plugin.cu]: 遗漏了, 已经删除, 感谢提醒

**Review by MARD1NO — APPROVED** [review]

**Review by heavengate — APPROVED** [review]

**Pull request merged**